### DISCUSSION TITLE
Hi, the question about withdraw 25% 50% 75%

### DISCUSSION THREAD

**Discussion by miner007** [Rarity-Extended/RarityExtended#79] [Q&A] [answered]
Since the UI in your wed has only deposit and I cannot find withdraw button. So, I read the code, please let me understand how to do a withdrawal directly via smart contract.  
In the path 'RarityExtended/pages/town/tavern/index.js' row 195, the apeInVault has connected to https://ftmscan.com/address/0xfCE6CbeF3867102da383465cc237B49fF4B9d48F.

I am appreciate your work, I am facing an invalid number value (arg="amount", coderType="uint256", value="9.9") situation when I write the smart contract. 

1. Can you provide me a data structure for writing it?
2. Where can I read the balance? 

thanks.

**Comment by saltyfacu** [comment, marked as answer]
Hello @miner007! We've updated the bank and now you can withdraw from there. Please try and come back to tell us how it went :)

**Comment by miner007** [comment]
Hi, the bank is great, the ui & ux are excellent.
